src/app/common/hooks/analytics/use-analytics.ts

```diff
@@ -38,6 +40,8 @@ export function useAnalytics() {
 
     return {
       async page(...args: PageParams) {
+        if (hasDeclined) return;
```

Comment:
This should come after the `logger.info` section, otherwise we won't build local logs